### ISSUE TITLE
Why aren't plugin logs gathered?

### ISSUE THREAD

**Issue by johnSchnake** [vmware-tanzu/sonobuoy#1415]
**What steps did you take and what happened:**
Run a plugin; check results tarball for pod logs and you typically only find the `sonobuoy` aggregator in the sonobuoy namespace. Where are the plugins?

They're already cleaned up.

I noticed this a few times in the past but I sweat it didn't used to be like that. Potentially it was an unrealized race condition? We should wait to cleanup the pods until we get the logs, not just exit as soon as it finishes. I think this is an option on the sonobuoy worker sidecar.

**Comment by johnSchnake** [comment]
FWIW there is a TL;DR; at the end but this comment represents my repro/debugging of the issue and it is my hope that if we break something related to this we may find these messages helpful.

You can see that the plugin cleanup occurs late in the game after the logs are queried:

https://github.com/vmware-tanzu/sonobuoy/blob/dc4be9cc83fa835fc84d78eb9c968a7ac1ae2044/pkg/discovery/discovery.go#L215

That means the issue must be in the pod letting itself terminate before then.

Found that logic here and...preseto I found I broke it in this commit https://github.com/vmware-tanzu/sonobuoy/pull/1072
https://github.com/vmware-tanzu/sonobuoy/pull/1072/files#diff-c478e3b28013044b4e1cf65ed2a0418986f28e1451774558b1e515583cb2e3abR139-R151

I must have just kept checking that 'it worked' and we just didnt have any tests to ensure that plugin logs were included.

A little mystery remains because when I tailed the logs to test this I saw:
```
... Sleeping forever...
```

but the worker code shows it doesnt pass the sleep value: https://github.com/vmware-tanzu/sonobuoy/blob/37b2df1e85fb5558c9f94ba139aa057efffff90e/pkg/plugin/driver/job/job.go#L119

Ah, but it seems that daemonset worker definition _does_ pass the sleep parameter as intended: https://github.com/vmware-tanzu/sonobuoy/blob/37b2df1e85fb5558c9f94ba139aa057efffff90e/pkg/plugin/driver/daemonset/daemonset.go#L161

TL;DR; I broke it. It will be a tiny fix. It was confusing because it was only wrong half the time. Welcome to my TED talk.

**Comment by johnSchnake** [comment]
Mystery deepends; that PR didnt remove a script on how the job plugin worker was invoked at all; just the daemonsets. So was it already broken at that point??? I know the fix now but I really need to know when/how it broke.

**Comment by johnSchnake** [comment]
For the life of me I can't find it. I dont recall the mechanism by which the pods hung around for jobs. I never see sleep being sent to them in code or in scripts (even ones that have since been deleted). I dont see any comments about it and I'm looking for all historical cases of where we were calling "sonobuoy worker" or "sleep" or "wait" throughout the codebase.

I guess it'll just be a mystery and I'll fix it as is.

**Comment by johnSchnake** [comment]
I FIGURED IT OUT! https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1164-remove-selflink

SelfLink was used as a key to map the pods so that we didnt iterate over them in 2 different places. After deprecation, that value just returns as empty so after the aggregator gets read, all the others are considered the same pod (since they have the same SelfLink) and skipped.
https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1164-remove-selflink
